overlord/snapshotstate/backend/backend.go

```diff
@@ -476,9 +476,51 @@ func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer,
 
 var ErrCannotCancel = errors.New("cannot cancel: import already finished")
 
+// multiError is a helper to handle mutliple errors.
+type multiError struct {
+	header string
+	errs   []error
+}
+
+// newMultiError returns a new multiError struct initialized with
+// the given format string that explains what operation potentially
+// want wrong. multiError can be nested and will render correctly
+// in these cases.
+func newMultiError(header string, errs []error) error {
+	return &multiError{header: header, errs: errs}
+}
+
+// Error formats the error string.
+func (me *multiError) Error() string {
+	return me.nestedError(0, make(map[error]struct{}))
+}
+
+// helper to ensure formating of nested multiErrors works.
+func (me *multiError) nestedError(level int, seen map[error]struct{}) string {
```

Comment:
do we really need seen? it seems a bit unlikely that somebody would setup a circular situation, if we are really worried, we could instead have a limit on level after which we print some generic error "more than expected nested errors"

Reply:
A good question. I mostly added it as it's a source of frustration if by accident it somehow ends up being circular. I can simplify it by just adding a max nesting level. Curious, what do you dislike about the check?

Reply:
it feels a bit overkill for here now that is all unexported.  OTOH if we tried to generalize this I'm not sure it would look nice, because as I said we would need to introduce an interface that corresponds to nestedError availability.